enterprise/server/backends/distributed/distributed.go

```diff
@@ -721,16 +720,20 @@ func (c *Cache) FindMissing(ctx context.Context, resources []*rspb.ResourceName)
 			}
 			continue
 		}
+
 		if len(foundMap) == len(hashResources) {
-			// If we've found everything, we can exit now.
+			// If we've tried everything, we can exit now.
 			break
 		}
 	}
 
 	// For every digest we found, if we did not find it
 	// on the first peer in our list, we want to backfill it.
 	backfills := make([]*backfillOrder, 0)
```

Comment:
does backfilling have a point anymore if we're only looking at the first successful response? We can try more than one peer if a peer errors out but that doesn't mean a backfill is necessary.

Reply:
I was trying to minimize the size of this change but I agree, will remove it.